ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,292 @@
+<?php
+/**
+ * @file
+ * E-mail reports functionality.
+ */
+
+
+/**
+ * Form settings mailing reports.
+ *
+ * @return
+ *   Save results form settings.
+ */
+function ding_dibs_setting_emails($form, &$form_state) {
+  drupal_set_title(t('Ding DIBS setting emails'));
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $hour = range(0, 23);
+  $options_hour = array_map(generate_zerro, $hour);
+  $minute = range(0, 59);
+  $options_minute = array_map(generate_zerro, $minute);
+  $text = '<label for="edit-ding-dibs-start-cron">';
+  $text .= t('Set the daily hour to run this cron');
+  $text .= '<span title="' . t('This field is required.');
+  $text .= '" class="form-required">*</span></label>';
+  $form = array();
+  $form['ding_dibs_start_cron'] = array(
+    '#type' => 'select',
+    '#options' => $options_hour,
+    '#default_value' => variable_get('ding_dibs_start_cron', 2),
+    '#required' => TRUE,
+    '#prefix' => $text . '<div class="container-inline">',
+  );
+  $form['ding_dibs_start_cron_minute'] = array(
+    '#type' => 'select',
+    '#options' => $options_minute,
+    '#default_value' => variable_get('ding_dibs_start_cron_minute', 0),
+    '#required' => TRUE,
+    '#suffix' => '</div><div class="description">' . t('Set the daily hh:ii to run this cron.') . '</div>',
+  );
+  $form['ding_dibs_emails'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Input emails that should be notified if something unusual happens with a payment.'),
+    '#description' => t('Enter up to ten email addresses Enter through the separator or comma, without blank lines.'),
+    '#default_value' => variable_get('ding_dibs_emails', ''),
+    '#required' => TRUE,
+    '#element_validate' => array('ding_dibs_emails_validate'),
+  );
+  return system_settings_form($form);
+}
+
+/**
+ * Validation forms element email report settings.
+ */
+function ding_dibs_emails_validate($element, &$form_state, $form) {
```

Comment:
The checking of empty row has been removed (https://github.com/ding2/ding_dibs/pull/4#discussion_r44968997), and it has been proposed that this check remains in order to not call the function which sends an email when the array has an empty row.